controllers/users.js

```diff
@@ -417,6 +417,31 @@ const addDefaultArchivedRole = async (req, res) => {
   }
 };
 
+/**
+ * Returns the lists of users who match the specified query params
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const filterUserBasedOnQuery = async (req, res) => {
```

Comment:
Does `BasedOnQuery` suffix in the name deliver additional value?

Reply:
`BasedOnQuery` suffix is not providing additional value to the function name. I will remove it.